api/post.go

```diff
@@ -405,6 +406,36 @@ func fireAndForgetNotifications(post *model.Post, teamId, teamUrl string) {
 					bodyPage.Props["PostMessage"] = model.ClearMentionTags(post.Message)
 					bodyPage.Props["TeamLink"] = teamUrl + "/channels/" + channel.Name
 
+					// attempt to fill in a message body if the post doesn't have any text
+					if len(strings.TrimSpace(bodyPage.Props["PostMessage"])) == 0 && len(post.Filenames) > 0 {
+						// extract the filenames from their paths and determine what type of files are attached
+						filenames := make([]string, len(post.Filenames))
+						onlyImages := true
+						for i, filename := range post.Filenames {
+							var err error
+							if filenames[i], err = url.QueryUnescape(filepath.Base(filename)); err != nil {
+								// this should never error since filepath was escaped using url.QueryEscape
+								filenames[i] = filepath.Base(filename)
+							}
+
+							ext := filepath.Ext(filename)
+							onlyImages = onlyImages && model.IsFileExtImage(ext)
+						}
+						filenamesString := strings.Join(filenames, ", ")
+
+						var attachmentPrefix string
+						if onlyImages {
+							attachmentPrefix = "Image"
+						} else {
+							attachmentPrefix = "File"
+						}
+						if len(post.Filenames) > 1 {
+							attachmentPrefix += "s"
```

Comment:
@coreyhulen This sort of thing is something that needs to be looked at for localization. I know we have this in the code in a number of places.